service/matching/matchingEngine.go

```diff
@@ -609,11 +612,8 @@ func (e *matchingEngineImpl) listTaskListPartitions(request *m.ListTaskListParti
 	if err != nil {
 		return nil, err
 	}
-	partitionHostInfo := make([]*workflow.TaskListPartitionMetadata, len(partitions))
 
-	if err != nil {
-		return nil, err
-	}
+	var partitionHostInfo []*workflow.TaskListPartitionMetadata
 	for _, partition := range partitions {
 		if host, err := e.getHostInfo(partition); err != nil {
```

Comment:
@anish531213 , @meiliang86 – I am curious about the **` !=`** in `err != nil`.

This blocks adds a new `TaskListPartitionMetadata` object when `getHostInfo()` *fails*, and I am unclear about the intention here.

Thank you for any insights you might have here!